### ISSUE TITLE
[Bug] NullReferenceException in the Shell on UWP

### ISSUE THREAD

**Issue by holecekp** [xamarin/Xamarin.Forms#12652]
### Description
I am experiencing random crashes in my Xamarin.Forms application on UWP. This specific crash occurred when I clicked on a flyout item in the shell menu to navigate to another page.

**System.NullReferenceException: Object reference not set to an instance of an object.**
**Stack trace** (from AppCenter):
<img width="604" alt="NRE_in_shell" src="https://user-images.githubusercontent.com/24872113/97539636-96d7b180-19c2-11eb-8dc2-22e680d4eba7.png">
<img width="600" alt="NRE_in_shell2" src="https://user-images.githubusercontent.com/24872113/97539642-99d2a200-19c2-11eb-9dd8-27f69a06d8f2.png">

**OS:** WINDOWS 10.0.18363
**Crashed after:** 18m 51s

### Steps to Reproduce

1. This issue occurs randomly so there are no steps to reproduce it. I have selected the same flyout item multiple times before and it did not crash.

### Expected Behavior
The application is stable and does not crash.

### Actual Behavior
The debug version is stable but the release version downloaded from Microsoft Store crashes on some occasions.

### Basic Information

- Version with issue: 4.8.0.1560
- Last known good version: ?
- IDE: VS 2019
- Platform Target Frameworks: 10.0.18362.0 (TargetPlatformVersion from csproj file)
  - UWP:   
- Nuget Packages:
    Microsoft.AppCenter.Crashes (3.4.3)
    Microsoft.NETCore.UniversalWindowsPlatform (6.2.10)
    Xamarin.Essentials (1.5.3.2)
    Xamarin.Forms (4.8.0.1560)

### Reproduction Link

This issue occurs randomly so it is not possible to create a project for reproducing it.

 ### Workaround

No workaround

**Comment by PureWeen** [comment]
@holecekp can you test on 5.0.0 pre3 and let me know if you see the same exception?

**Comment by holecekp** [comment]
@PureWeen Yes, after about 30 minutes of trying, I was able to reproduce this crash also in 5.0.0 pre 3.

**Comment by PureWeen** [comment]
@holecekp do you have a repro I can play with?

**Comment by holecekp** [comment]
@PureWeen  I have created a repro and I was able to reproduce the NRE for 4 times in it.
[UwpShellTestProject.zip](https://github.com/xamarin/Xamarin.Forms/files/5469037/UwpShellTestProject.zip)

**Steps:**
This bug is very hard to reproduce. It requires a lot of patience. I took me 3 - 20 minutes of random clicking to crash the application. There are some steps that I did randomly.
- Selecting different flyout items in the shell menu.
- There are two tabs in one of the flyout items (Morse Codes and Q-Codes List). I have also browsed between them.
- There is a button on the About that is used for navigating to another page that is not accessible directly from the shell menu. I have used the button and returned back to the About page.
- I was changing windows size. By the way, this demonstrates another Xamarin.Forms bug on UWP. When you select another page from shell menu, resize the window, and return to the previous page, the layout is broken (it corresponds to the previous window size).
- On the settings page, there are various controls. I have scrolled on the page, clicked on the controls and clicked on the space between them (this again shows another bug, the ScrollView position is reseted on UWP in these cases).
- I was switching between light and dark theme in Windows settings.
- I was clicking randomly on various controls on the pages.

After some time, when I click on a flyout item in the shell menu, the application crashed. It was different flyout item every time. It seems that there is a great deal of randomness.

I am including screen shots. This time I got only message without a stack trace. I have subscribed App.UnhandledException event in the UWP project but it did not helped (still no stack trace). I have enabled debugging also in native code and tried to load required symbols but I only got message that required symbols could not be loaded (again, screenshot is included).

I have used another computer so I can rule out the case that this would be a problem only of a single specific computer (corrupted system libraries, configuration, etc.).

**Screenshots:**
![XF UWP shell NRE](https://user-images.githubusercontent.com/24872113/97779163-a4d13200-1b7c-11eb-83e2-7282a0f55451.png)
![XF UWP shell NRE debug native code](https://user-images.githubusercontent.com/24872113/97779166-a864b900-1b7c-11eb-8256-80897d1988b4.png)
